src/Core/Grid/Definition/Factory/MerchandiseReturnProductsGridDefinitionFactory.php

```diff
@@ -0,0 +1,255 @@
+                    ])
+            )
+        ;
+
+        return $columns;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getFilters()
+    {
+        return (new FilterCollection())
+            ->add((new Filter('reference', TextType::class))
+            ->setTypeOptions([
+                'required' => false,
+                'attr' => [
+                    'placeholder' => $this->trans('Search Reference', [], 'Admin.Actions'),
+                ],
+            ])
+            ->setAssociatedColumn('product_reference')
+            )
+            ->add((new Filter('product_name', TextType::class))
+            ->setTypeOptions([
+                'required' => false,
+                'attr' => [
+                    'placeholder' => $this->trans('Search Name', [], 'Admin.Actions'),
+                ],
+            ])
+            ->setAssociatedColumn('product_name')
+            )
+            ->add((new Filter('quantity', TextType::class))
+            ->setTypeOptions([
+                'required' => false,
+                'attr' => [
+                    'placeholder' => $this->trans('Search quantity', [], 'Admin.Actions'),
+                ],
+            ])
+            ->setAssociatedColumn('quantity')
+            )
+            ->add((new Filter('actions', SearchAndResetType::class))
+            ->setTypeOptions([
+                'reset_route' => 'admin_common_reset_search_by_filter_id',
+                'reset_route_params' => [
+                    'filterId' => self::GRID_ID,
+                    'merchandiseReturnId' => $this->merchandiseReturnId,
+                ],
+                'redirect_route' => 'admin_merchandise_returns_edit',
+                'redirect_route_params' => [
+                    'merchandiseReturnId' => $this->merchandiseReturnId,
+                ],
+            ])
+            ->setAssociatedColumn('actions')
+            )
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getBulkActions()
+    {
+        return (new BulkActionCollection())
+            ->add(
+                (new SubmitBulkAction('delete_selection'))
+                    ->setName($this->trans('Delete selected', [], 'Admin.Actions'))
+                    ->setOptions([
+                        'submit_route' => 'admin_merchandise_returns_delete_product_bulk',
+                        'route_params' => [
+                            'merchandiseReturnId' => $this->merchandiseReturnId,
+                        ],
+                        'confirm_message' => $this->trans('Are you sure you want to delete the selected item(s)?', [], 'Admin.Global'),
+                        'modal_options' => new ModalOptions([
+                            'title' => $this->trans('Delete selection', [], 'Admin.Actions'),
+                            'confirm_button_label' => $this->trans('Delete', [], 'Admin.Actions'),
+                            'confirm_button_class' => 'btn-danger',
```

Comment:
Shouldn't it be a cancel button as well?

Reply:
This is a delete button for a merchandise return product. In the old form, there is only action to delete product from merchandise return. 

![image](https://user-images.githubusercontent.com/47781757/86892713-45e93800-c109-11ea-8567-0b8d7f129ca5.png)

If you mean cancel button for the whole page then there is one in the form. Form and list are both part of the same page, I am not sure if the list requires the cancel button of its own as well. What do you think @LouiseBonnard 

![image](https://user-images.githubusercontent.com/47781757/86892852-78933080-c109-11ea-9317-4ea08802e6c9.png)

Reply:
My bad, I thought it was a modal. We should keep it iso then, thanks a lot!